src/fixate/examples/jig_driver.py

```diff
@@ -58,3 +58,34 @@ class JigMuxGroup(MuxGroup):
 jig.mux.mux_two("sig5")
 jig.mux.mux_three("On")
 jig.mux.mux_three(False)
+
+try:
+    from typing import Annotated
+except ImportError:
+    # 3.8
+    from typing_extensions import Annotated
+
+from typing import Literal, Union
+
+# maybe we can create aliases to make it easier to understand how to create a MuxDef
+SignalName = Literal
+Signal = Annotated
+MuxDef = Union
```

Comment:
I think it is probably worth including these in the module and we would document the usage of the class using them.

If nothing else, it will make grepping for places where they are used easier. However, There is a class with the name `Signal`. Perhaps:

```python
SignalName = Literal
SignalDef = Annotated
MuxDef = Union

MuxOneSigDef = MuxDef[
        SignalDef[SignalName["sig_a1"], "a0", "a2" ], 
        SignalDef[SignalName["sig_a2"], "a1"]
    ]
```

Reply:
Grepping would be a good use to bother to do this, however extending the behaviour of this classes is basically prevented by a bunch of runtime bits.
Since these are types and not objects it makes it really hard to change the behaviour e.g debug messages will still print the original typing names.

Reply:
I thought this is what you meant earlier today with aliases.
Mypy doesn't like assigning things like `Literal` to a variable. Assigning the mux definition to a variable is ok since that's a properly defined type.